src/types.rs

```diff
@@ -26,6 +29,89 @@ macro_rules! unimplemented {
     };
 }
 
+#[derive(Clone, Debug)]
+pub struct GasCosts {
+    pub operator_cost: u64,
+    pub launch_cost: u64,
+    pub abi_costs: HashMap<String, u64>,
```

Comment:
Maybe we should not set this as pub to avoid dev error?

Reply:
Done.